vtex-sites/gatsby.store
# Feat: Newsletter form and useNewsletter hook

## What's the purpose of this pull request?

Adds a newsletter form and a hook that uses the [subscribeToNewsletter mutation](https://github.com/vtex/faststore/pull/1385).

## How does it work?

It uses the `subscribeToNewsletter` mutation to add the person's info to the store's newsletter list.

## How to test it?

The Newsletter form is in the footer, fill in the form fields and then check if it was saved on MasterData using this GET:
https://storeframework.vtexcommercestable.com.br/api/dataentities/NL/search?email=HERE
- Substitute _HERE_ with the email filled in the form
- Add a cookie VtexIdclientAutCookie with your auth token

## References

<em>Spread the knowledge: is this any content you used to create this PR that is worth sharing?</em>

<em>Extra tip: add references to related issues or mention people important to this PR may be good for the documentation and reviewing process</em>

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**

**PR Description**

**Documentation**
